lib/src/core/common/widgets/customized_appbar.dart

```diff
@@ -0,0 +1,89 @@
+import 'package:flutter/material.dart';
+import 'package:window_manager/window_manager.dart';
+
+class CustomAppBar extends StatelessWidget implements PreferredSizeWidget {
+  const CustomAppBar({super.key});
+
+  @override
+  Size get preferredSize => const Size.fromHeight(48);
+
+  @override
+  Widget build(BuildContext context) {
+    return GestureDetector(
+      onPanStart: (details) async {
+        await windowManager.startDragging();
+      },
+      child: Container(
+        height: preferredSize.height,
+        padding: const EdgeInsets.symmetric(horizontal: 10),
+        decoration: BoxDecoration(
+          color: Colors.grey[900],
+        ),
+        child: Row(
+          children: [
+            Expanded(
+              child: Align(
+                alignment: AlignmentDirectional.centerStart,
+                child: Text(
+                  'Linux-like AppBar',
+                  style: TextStyle(color: Colors.white, fontSize: 16),
+                ),
+              ),
+            ),
+            Row(
+              children: [
+                _AppBarButton(
+                  icon: Icons.remove,
+                  color: Colors.yellow,
+                  onPressed: () async {
+                    await windowManager.minimize();
+                  },
+                ),
+                _AppBarButton(
+                  icon: Icons.crop_square,
+                  color: Colors.green,
+                  onPressed: () async {
+                    final isMaximized = await windowManager.isMaximized();
+                    if (isMaximized) {
+                      await windowManager.unmaximize();
+                    } else {
+                      await windowManager.maximize();
+                    }
+                  },
+                ),
+                _AppBarButton(
+                  icon: Icons.close,
+                  color: Colors.red,
+                  onPressed: () async {
+                    await windowManager.close();
+                  },
```

Comment:
The color of the button is not correct. Please change follow the design
Light mode: #242424 
Dark mode: #FFFFFF

Reply:
These colors should be defined and managed within a color palette and text styles. 
The application should support switching between light and dark themes.

**Dependencies:**

* This feature depends on the resolution of Pull Request #45.
* Access to the Figma design system is required.